data-preprocessing/video-download-upload/video-download-upload.py

```diff
@@ -0,0 +1,145 @@
+        print("JSON file contains all the required keys")
+    except ValueError as e:
+        print(str(e))
+
+    #read json file
+    jsonData = readInstructionFile()
+
+    #Define doc variables
+    excelDocumentName = jsonData['excel_doc_path']  #document path
+    excelSheetName = jsonData['sheet_name']  #name of sheet in document
+    videosHeaderName = jsonData['video_header_name'] #name of header of column containing video links
+    localVideoDirectoryPath = jsonData['video_folder_path'] #path to folder that will contain downloaded videos locally
+    serviceAccountFilePath = jsonData['service_account_path'] #path to service account file 
+
+    #check if local folder exists. !folder_exists? create it: continue
+    if not os.path.exists(localVideoDirectoryPath):
+        os.makedirs(localVideoDirectoryPath)
+        print(f"Folder '{localVideoDirectoryPath}' does not exist. Folder created successfully")
+    else:
+        print(f"Folder '{localVideoDirectoryPath}' already exists.")
+
+
+    # Read the Excel sheet and extract video links
+    df =readExcelSheet(excelDocumentName, excelSheetName)
+    videoLinks = df[videosHeaderName]
+
+    #Loop through list and download each video
+    for link in videoLinks:
+        downloadVideo(link, localVideoDirectoryPath)
+
+    # Authenticate with Google Drive API and create a folder to store the videos
+    creds = service_account.Credentials.from_service_account_file(serviceAccountFilePath)
+    driveService = build('drive', 'v3', credentials=creds)
+
+    #name of local folder to contain donwload videos
+    folder_name = jsonData['google_drive_folder_name']
+
+    fileMetaData = {'name': folder_name, 'mimeType': 'video/mp4'}
+    folder = driveService.files().create(body=fileMetaData, fields='id').execute()
+    folder_id = folder.get('id') #can also be picked up from jsonData
+    #folder_id = jsonData['google_drive_folder_ID']
+
+    print(folder_id)
+
+
+    # Upload the videos to the Google Drive folder
+    for videoPath in Path(localVideoDirectoryPath).glob('*.mp4'):
+        try:
+            print(f"Uploading '{videoPath}' from '{localVideoDirectoryPath}'")
+            fileMetaData = {'name': videoPath.name, 'parents': [folder_id]}
+            media = MediaFileUpload(str(videoPath))
+            file = driveService.files().create(body=fileMetaData, media_body=media, fields='id').execute()
+            print(f'Successfully uploaded {videoPath} to {folder_name} folder.') 
+        except HttpError as error:
+            print(f"An error occurred: '{error}' on '{videoPath}'")
+
+
+#Main program entry point
+if __name__ == "__main__":
+    with open("README.md", "r") as f:
+        description = f.read()
+
+    parser = argparse.ArgumentParser(description=description)
+    parser.add_argument("--h", action="store_true", help=description)
+    args = parser.parse_args()
+
+    if args.h:
+        usage()
+    else:
+        __main__(args)
+        
```

Comment:
still seeing this issue. At the end of the file hit enter to create a new line and add that to the changes you push up and it should resolve it